File: src/backend/agenda/src/model/view_pessoa_usuario_model.py
from src import db


class ViewPessoaUsuarioModel(db.Model):
    __tablename__ = 'view_pessoa_usuario'

    id = db.Column(db.Integer, primary_key=True)
    id_pessoa = db.Column(db.Integer)
    pessoa_nome = db.Column(db.String(450))
    tipo = db.Column(db.String(3))
    email = db.Column(db.String(750))
    id_colaborador = db.Column(db.Integer)
    id_usuario = db.Column(db.Integer)
    login = db.Column(db.String(150))
    senha = db.Column(db.String(150))
    data_cadastro = db.Column(db.DateTime)
    administrador = db.Column(db.String(3))


    def mapping(self, data):
        self.id = data.get('id')
        self.id_pessoa = data.get('idPessoa')
        self.pessoa_nome = data.get('pessoaNome')
        self.tipo = data.get('tipo')
        self.email = data.get('email')
        self.id_colaborador = data.get('idColaborador')
        self.id_usuario = data.get('idUsuario')
        self.login = data.get('login')
        self.senha = data.get('senha')
        self.data_cadastro = data.get('dataCadastro')
        self.administrador = data.get('administrador')

    def serialize(self):
        return {
            'id': self.id,
            'idPessoa': self.id_pessoa,
            'pessoaNome': self.pessoa_nome,
            'tipo': self.tipo,
            'email': self.email,
            'idColaborador': self.id_colaborador,
            'idUsuario': self.id_usuario,
            'login': self.login,
            'senha': self.senha,
            'dataCadastro': self.data_cadastro.isoformat(),
            'administrador': self.administrador,
        }

File: src/frontend/wms/lib/app/data/model/view_pessoa_usuario_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:wms/app/infra/infra_imports.dart';
import 'package:intl/intl.dart';

class ViewPessoaUsuarioModel {
	int? id;
	int? idPessoa;
	String? pessoaNome;
	String? tipo;
	String? email;
	int? idColaborador;
	int? idUsuario;
	String? login;
	String? senha;
	DateTime? dataCadastro;
	String? administrador;

	ViewPessoaUsuarioModel({
		this.id,
		this.idPessoa,
		this.pessoaNome,
		this.tipo,
		this.email,
		this.idColaborador,
		this.idUsuario,
		this.login,
		this.senha,
		this.dataCadastro,
		this.administrador,
	});

	static List<String> dbColumns = <String>[
		'id',
		'id_pessoa',
		'pessoa_nome',
		'tipo',
		'email',
		'id_colaborador',
		'id_usuario',
		'login',
		'senha',
		'data_cadastro',
		'administrador',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Id Pessoa',
		'Pessoa Nome',
		'Tipo',
		'Email',
		'Id Colaborador',
		'Id Usuario',
		'Login',
		'Senha',
		'Data Cadastro',
		'Administrador',
	];

	ViewPessoaUsuarioModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idPessoa = jsonData['idPessoa'];
		pessoaNome = jsonData['pessoaNome'];
		tipo = jsonData['tipo'];
		email = jsonData['email'];
		idColaborador = jsonData['idColaborador'];
		idUsuario = jsonData['idUsuario'];
		login = jsonData['login'];
		senha = jsonData['senha'];
		dataCadastro = jsonData['dataCadastro'] != null ? DateTime.tryParse(jsonData['dataCadastro']) : null;
		administrador = jsonData['administrador'];
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idPessoa'] = idPessoa;
		jsonData['pessoaNome'] = pessoaNome;
		jsonData['tipo'] = tipo;
		jsonData['email'] = email;
		jsonData['idColaborador'] = idColaborador;
		jsonData['idUsuario'] = idUsuario;
		jsonData['login'] = login;
		jsonData['senha'] = senha;
		jsonData['dataCadastro'] = dataCadastro != null ? DateFormat('yyyy-MM-ddT00:00:00').format(dataCadastro!) : null;
		jsonData['administrador'] = administrador;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idPessoa = plutoRow.cells['idPessoa']?.value;
		pessoaNome = plutoRow.cells['pessoaNome']?.value;
		tipo = plutoRow.cells['tipo']?.value;
		email = plutoRow.cells['email']?.value;
		idColaborador = plutoRow.cells['idColaborador']?.value;
		idUsuario = plutoRow.cells['idUsuario']?.value;
		login = plutoRow.cells['login']?.value;
		senha = plutoRow.cells['senha']?.value;
		dataCadastro = Util.stringToDate(plutoRow.cells['dataCadastro']?.value);
		administrador = plutoRow.cells['administrador']?.value;
	}	

	ViewPessoaUsuarioModel clone() {
		return ViewPessoaUsuarioModel(
			id: id,
			idPessoa: idPessoa,
			pessoaNome: pessoaNome,
			tipo: tipo,
			email: email,
			idColaborador: idColaborador,
			idUsuario: idUsuario,
			login: login,
			senha: senha,
			dataCadastro: dataCadastro,
			administrador: administrador,
		);			
	}

	
}